src/backup.cc

```diff
@@ -155,7 +155,20 @@ Backup::Backup(const Napi::CallbackInfo& info) : Napi::ObjectWrap<Backup>(info)
     }
 
     Database* db = Napi::ObjectWrap<Database>::Unwrap(info[0].As<Napi::Object>());
-    Napi::String filename = info[1].As<Napi::String>();
+    Database* otherDb = NULL;
+
+    Napi::String filename;
+
+    if (info[1].IsObject()) {
+        // A database instance was passed instead of a filename
+        otherDb = Napi::ObjectWrap<Database>::Unwrap(info[1].As<Napi::Object>());
+        otherDb->Ref();
```

Comment:
I am not sure if the Ref() call is needed here. I have applied this from @paulfitz.
I assume it is needed because it prevents the otherDb instance to be freed while a backup is in progress, which should usually not happen unless the user closes the db during a backup process.

Please someone double check.